Compile fails with target_cpu on GitHub Actions for OSX - illegal instruction

When compiling  a project using GitHub Actions for OSX and targeting a CPU other than ```native```, the compile fails with
```
error: failed to run custom build command for `typenum v1.12.0`

Caused by:
  process didn't exit successfully: `/Users/runner/work/tari/tari/target/release/build/typenum-4114bf614ff41b71/build-script-main` (signal: 4, SIGILL: illegal instruction)
--- stdout
cargo:rustc-env=TYPENUM_BUILD_CONSTS=/Users/runner/work/tari/tari/target/release/build/typenum-e6ec2a2a9a37d166/out/consts.rs

warning: build failed, waiting for other jobs to finish...
error: build failed
Error: Process completed with exit code 101.
```
Same project works locally fine for OSX and GitHub Action build for Ubuntu and Windows.  
  
Investigation shows that GHA OSX is using ```ivybridge``` hardware, where I have ```skylake``` and GHA Ubuntu and Windows hardware is been reported as ```skylake-avx512```

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -6,6 +6,7 @@ The MSRV (Minimum Supported Rust Version) is 1.37.0, and typenum is tested
 against this Rust version.
 
 ### Unreleased
+- [fixed] Cross-compilation issue due to doing math in build script.
 
 ### 1.14.0 (2021-09-01)
 - [changed] Sealed all marker traits. Documentation already stated that these

--- a/build/main.rs
+++ b/build/main.rs
@@ -80,8 +80,10 @@ pub fn no_std() {}
 fn main() {
     let highest: u64 = 1024;
 
-    let first2: u32 = (highest as f64).log(2.0).round() as u32 + 1;
-    let first10: u32 = (highest as f64).log(10.0) as u32 + 1;
+    // Use hardcoded values to avoid issues with cross-compilation.
+    // See https://github.com/paholg/typenum/issues/162
+    let first2: u32 = 11; // (highest as f64).log(2.0).round() as u32 + 1;
+    let first10: u32 = 4; // (highest as f64).log(10.0) as u32 + 1;
     let uints = (0..(highest + 1))
         .chain((first2..64).map(|i| 2u64.pow(i)))
         .chain((first10..20).map(|i| 10u64.pow(i)));

